pandas/core/resample.py

```diff
@@ -1519,8 +1519,26 @@ def _get_period_bins(self, ax):
 
         start = ax.min().asfreq(self.freq, how=self.convention)
         end = ax.max().asfreq(self.freq, how='end')
-
-        labels = binner = PeriodIndex(start=start, end=end,
+        if self.base:
+            start = start.to_timestamp()
+            end = end.to_timestamp()
+
+            # get base adjusted bin edge labels
+            p_start, p_end = _get_range_edges(start,
+                                              end,
+                                              self.freq,
+                                              closed=self.closed,
+                                              base=self.base)
+
+            # compensate if edge labels are extened away from true labels
```

Comment:
can you push this into _get_range_edges w/o breaking the world?

Reply:
Done. also added some tests to cover the start date not landing onOffset